packages/contracts/scripts/one-off/add-unregistered-contracts.ts

```diff
@@ -0,0 +1,236 @@
+// CONFIG BEGINS HERE
+//////////////////////////////////////////////////////////////////////////////
+type BaseConfig = {
+  network: string;
+  engineFactoryAddress: string;
+  useLedgerSigner: boolean;
+};
+
+type GnosisSafeConfig = BaseConfig & {
+  useGnosisSafe: true;
+  safeAddress: string;
+  transactionServiceUrl: string;
+};
+
+type NoGnosisSafeConfig = BaseConfig & {
+  useGnosisSafe: false;
+  safeAddress?: never;
+  transactionServiceUrl?: never;
+};
+
+type Config = GnosisSafeConfig | NoGnosisSafeConfig;
+
+const config: Config = {
+  network: "mainnet",
+  engineFactoryAddress: "0x000000004058B5159ABB5a3Dd8cf775A7519E75F",
+  useLedgerSigner: false,
+  useGnosisSafe: true,
+  safeAddress: "0x52119BB73Ac8bdbE59aF0EEdFd4E4Ee6887Ed2EA",
+  transactionServiceUrl: "https://safe-transaction-mainnet.safe.global/",
+};
+
+//////////////////////////////////////////////////////////////////////////////
+// CONFIG ENDS HERE
+//////////////////////////////////////////////////////////////////////////////
+
+async function main() {
+  const networkName = await getNetworkName();
+
+  if (networkName !== config.network) {
+    throw new Error(
+      `network name ${networkName} does not match expected network name ${config.network}`
+    );
+  }
+
+  if (
+    config.useGnosisSafe &&
+    (!config.safeAddress || !config.transactionServiceUrl)
+  ) {
+    throw new Error(
+      `Must provide both safeAddress and transactionServiceUrl when using gnosis safe`
+    );
+  }
+
+  let signer: SignerWithAddress | Wallet;
+  if (config.useLedgerSigner) {
+    const ledgerAddress = hre.network.config.ledgerAccounts[0];
+    signer = await ethers.getSigner(ledgerAddress);
+    console.log("using wallet", await signer.getAddress());
+  } else {
+    const deployerWallet = getDeployerWallet();
+
+    if (!deployerWallet) {
+      throw new Error("Deployer wallet not found");
+    }
+
+    signer = deployerWallet.connect(ethers.provider);
+  }
+
+  let gnosisSetup: {
+    ethAdapter: EthersAdapter;
+    safeApiKit: SafeApiKit;
+    protocolKit: Safe;
+  } | null = null;
+
+  // Gnosis sdk setup
+  if (config.useGnosisSafe) {
+    const ethAdapter = new EthersAdapter({
+      ethers,
+      signerOrProvider: signer,
+    });
+
+    const safeApiKit = new SafeApiKit({
+      txServiceUrl: config.transactionServiceUrl,
+      ethAdapter,
+    });
+
+    const protocolKit = await Safe.create({
+      ethAdapter,
+      safeAddress: config.safeAddress,
+    });
+
+    gnosisSetup = {
+      ethAdapter,
+      safeApiKit,
+      protocolKit,
+    };
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // ACTION BEGINS HERE
+  //////////////////////////////////////////////////////////////////////////////
+
+  // Query for unregistered contracts
+  const registryAddresses = [
+    "0x2ee7b9bb2e038be7323a119701a191c030a61ec6",
+    "0x652490c8bb6e7ec3fd798537d2f348d7904bbbc2",
```

Comment:
I think we ONLY want `0x2ee7b9bb2e038be7323a119701a191c030a61ec6` - `0x652490c8bb6e7ec3fd798537d2f348d7904bbbc2` is legacy, and won't be checked by the dependency registry